core/src/main/java/com/opensymphony/xwork2/ognl/OgnlUtil.java

```diff
@@ -82,7 +86,18 @@ public void setXWorkConverter(XWorkConverter conv) {
 
     @Inject(StrutsConstants.STRUTS_DEVMODE)
     public void setDevMode(String mode) {
-        this.devMode = BooleanUtils.toBoolean(mode);
+        final boolean booleanMode = BooleanUtils.toBoolean(mode);
```

Comment:
Why? If I remember correctly then injection can happen multiple times and some plugins, for example, can override already injected properties. That is how some things work (`disallowProxyMemberAccess` is one of this things).

Reply:
@apache/struts-committers , I think here @JCgH4164838Gh792C124B5 is trying to prevent hacker from the ability of resetting those values that menaces overall security. Whether it works or breaks here, In general it seems a nice devised practice :+1: thank you!

Reply:
@yasserzamani How it is helpful if basic stuff isn't working?

If this is really the case then they can be protected in some other way.

Reply:
Yeah basic stuff should work. I actually meant the idea itself - to prevent hacker from the ability of resetting those values that menaces overall security.

Reply:
Hello Struts Team.

The primary intention for this PR was only to try and limit the ability to change certain elements (such as devMode), for which there didn't seem to be a good reason for them to change after the 1st (initial) configuration.  

It was more aimed at avoiding inadvertent/accidental state changes for important state than anything else.

None of the regression tests, nor interaction with the Showcase application encountered any issues so it seemed like a reasonable approach to take.

If there are certain properties (such as `disallowProxyMemberAccess`) for which subsequent changes are necessary/valid, then they could be omitted by this or another mechanism that attempts to limit such changes.

Reply:
Hello @aleksandr-m  and @yasserzamani .  

Do either of you think _any_ of the properties I attempted to apply change-limiters to could/should have the change-limiter applied to them ?

I could eliminate some or all of the change-limiters in a revision of this PR (and PR #291) that only retains the additional logging around property changes.  Please let me know what you think.

Reply:
> Do either of you think any of the properties I attempted to apply change-limiters to could/should have the change-limiter applied to them ?

Well, currently I don't see what benefits does it bring.

Reply:
Hi @JCgH4164838Gh792C124B5 , I like your idea but to be frank, I'm not able to have a vision on future side effects.

But I remembered one of my old ideas that not only has somehow same benefits but also is lower risky and is more readable: **Refactoring all methods annotated with `@com.opensymphony.xwork2.inject.Inject` from `public setXXX` to `private injectXXX`.** This preserves those secure against OGNL access. As a glimpse, with IntelliJ IDE's refactoring tools, I did it on `OgnlUtil.setDevMode` and worked  :)